code/clue.js

```diff
@@ -27,38 +145,158 @@ const suspects = [
 // FINISH THIS FUNCTION TO SHOW ALL INFORMATION ABOUT THE KILLER.
-// This function will be invoked when you click on the killer card.
+
 const pickKiller = () => {
-  // This will randomly select a killer from the suspects. And add that to the mystery object.
   mystery.killer = randomSelector(suspects)
 
-  // This will change the background color of the card to the one connected to the chosen killer and show the full name of the killer.
   const theKiller = document.getElementById("killer")
   const theKillerName = document.getElementById("killerName")
 
   theKiller.style.background = mystery.killer.color
   theKillerName.innerHTML =
     mystery.killer.firstName + " " + mystery.killer.lastName
+
+
+  // REMAINING OBJECT INFO //
+  const theKillerAge = document.getElementById("killerAge")
+  theKillerAge.innerHTML =
+    mystery.killer.age
+
+  const theKillerDescription = document.getElementById("killerDescription")
+  theKillerDescription.innerHTML =
+    mystery.killer.description
+
+  const theKillerOccupation = document.getElementById("killerOccupation")
+  theKillerOccupation.innerHTML =
+    mystery.killer.occupation
+
+  const theKillerImage = document.getElementById("killerImage")
+  theKillerImage.src = mystery.killer.image
+
+}
+
+// FUNCTIONS //
+// weapons //
+const pickWeapon = () => {
+  mystery.weapon = randomSelector(weapons)
+
+  const theWeapon = document.getElementById("weaponName");
+  theWeapon.innerHTML = mystery.weapon.name
+
+  const theWeaponWeight = document.getElementById("weaponWeight")
+  theWeaponWeight.innerHTML = mystery.weapon.weight
+
+  const theWeaponMaterial = document.getElementById("weaponMaterial")
+  theWeaponMaterial.innerHTML = mystery.weapon.material
+
+  //Weapon images // 
+  const theWeaponImage = document.getElementById("weaponImage")
+  theWeaponImage.src = mystery.weapon.image
 }
 
-// CREATE FUNCTIONS pickWeapon and pickRoom in a similar way.
+// rooms //
+const pickRoom = () => {
+  mystery.room = randomSelector(rooms)
+
+  const theRoomName = document.getElementById("roomName");
+  theRoomName.innerHTML = mystery.room
+}
+
+// time of murder //
+const pickTime = () => {
+  mystery.time = randomSelector(times)
+
+  const theTimeName = document.getElementById("timeName");
+  theTimeName.innerHTML = mystery.time
+
+
+}
 
 // CREATE A FUNCTION revealMystery that will be invoked when you click that button. It should show something like:
 // 'The murder was committed by Jacob Green, in the living room with a rope.'
+
+const revealMystery = () => {
+  const theReveal = document.getElementById("mystery")
+
+  theReveal.innerHTML = (`The murder was committed by ${mysteryKiller.name} with a ${mysteryKiller.weapon} in the ${mysteryKiller.room} at ${mysteryKiller.time}.`);
+};
+
+const mysteryKiller = {
+  name: "mrs Peacock",
+  image: "assets/peacock.png",
+  weapon: "trophy",
+  room: "observatory",
+  time: "midnight"
+};
```

Comment:
Hm this is hardcoded values. Would be great if you can use you functions that you have been coding!

Reply:
I misunderstood so I will change asap.